src/OpenSpaceToolkit/Astrodynamics/Solvers/LeastSquaresSolver.cpp

```diff
@@ -0,0 +1,419 @@
+    ostk::core::utils::Print::Line(anOutputStream) << "Iteration Count: " << iterationCount_;
+    ostk::core::utils::Print::Line(anOutputStream) << "Termination Criteria: " << terminationCriteria_;
+
+    ostk::core::utils::Print::Separator(anOutputStream, "Solution State");
+    solutionState_.print(anOutputStream, false);
+
+    ostk::core::utils::Print::Separator(anOutputStream, "Steps");
+    for (const auto& step : steps_)
+    {
+        step.print(anOutputStream);
+    }
+
+    ostk::core::utils::Print::Footer(anOutputStream);
+
+    displayDecorator ? ostk::core::utils::Print::Footer(anOutputStream) : void();
+}
+
+Real LeastSquaresSolver::Analysis::getRmsError() const
+{
+    return rmsError_;
+}
+
+Size LeastSquaresSolver::Analysis::getIterationCount() const
+{
+    return iterationCount_;
+}
+
+String LeastSquaresSolver::Analysis::getTerminationCriteria() const
+{
+    return terminationCriteria_;
+}
+
+const State& LeastSquaresSolver::Analysis::accessSolutionState() const
+{
+    return solutionState_;
+}
+
+const MatrixXd& LeastSquaresSolver::Analysis::accessSolutionCovariance() const
+{
+    return solutionCovariance_;
+}
+
+const MatrixXd& LeastSquaresSolver::Analysis::accessSolutionFrisbeeCovariance() const
+{
+    return solutionFrisbeeCovariance_;
+}
+
+const Array<LeastSquaresSolver::Step>& LeastSquaresSolver::Analysis::accessSteps() const
+{
+    return steps_;
+}
+
+LeastSquaresSolver::LeastSquaresSolver(const Size& aMaxIterationCount, const Real& aRmsUpdateThreshold)
+    : maxIterationCount_(aMaxIterationCount),
+      rmsUpdateThreshold_(aRmsUpdateThreshold)
+{
+    if (aMaxIterationCount == 0)
+    {
+        throw ostk::core::error::RuntimeError("Max iteration count must be greater than 0.");
+    }
+
+    if (aRmsUpdateThreshold <= 0.0)
+    {
+        throw ostk::core::error::RuntimeError("RMS update threshold must be greater than 0.");
+    }
+}
+
+Size LeastSquaresSolver::getMaxIterationCount() const
+{
+    return maxIterationCount_;
+}
+
+Real LeastSquaresSolver::getRmsUpdateThreshold() const
+{
+    return rmsUpdateThreshold_;
+}
+
+LeastSquaresSolver::Analysis LeastSquaresSolver::solve(
+    const State& anInitialGuessState,
+    const Array<State>& aReferenceStateArray,
+    const std::function<Array<State>(const State&, const Array<Instant>&)>& aGenerateStatesCallback,
+    const std::unordered_map<CoordinateSubset, VectorXd>& anInitialGuessSigmas,
+    const std::unordered_map<CoordinateSubset, VectorXd>& aReferenceStateSigmas
+) const
+{
+    if (!anInitialGuessState.isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Initial guess state");
+    }
+
+    if (aReferenceStateArray.isEmpty())
+    {
+        throw ostk::core::error::runtime::Undefined("Reference states");
+    }
+
+    // Setup state builders
+    const StateBuilder estimationStateBuilder(anInitialGuessState);
+    const Shared<const Frame>& estimationStateFrame = anInitialGuessState.accessFrame();
```

Comment:
Estimation State Builder would use the estimationCoodinates, and we would nevertheless want a initialguessStateBuilder

```suggestion
    const Shared<const Frame>& estimationStateFrame = anInitialGuessState.accessFrame();
    const StateBuilder estimationStateBuilder(estimationStateFrame, estimationCoordinates);
    const StateBuilder initialGuessStateBuilder(anInnitialGuessState);
```